tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py

```diff
@@ -151,31 +172,116 @@ def _read_log_configs_from_master(self):
+        return [
+            log for log in logs if self.scheduler in log.get("schedulers") and self.platform in log.get("platforms")
+        ]
+
+    def _log_is_relevant_for_feature(self, log, node_config):
+        """Return a boolean describing whether log contains a feature_conditions entry relevant to self.feature_key."""
+        is_relevant_for_feature = False
+        for feature_condition in log.get("feature_conditions", []):
+            if all(
+                [
+                    self.feature_key == feature_condition.get("dna_key"),
+                    node_config.get(self.feature_key) in feature_condition.get("satisfying_values"),
+                ]
+            ):
+                is_relevant_for_feature = True
+                break
+        return is_relevant_for_feature
+
+    def _get_node_roles_for_which_feature_is_relevant(self, log, node_configs):
+        """
+        Return a list of the node types on which the feature-specific log is applicable.
+
+        This is necessary because even though a log might support a certain node role, that particular node type might
+        not be enabled on that node type in the current cluster.
+        """
+        applicable_node_roles = []
+        for node_role in log.get("node_roles"):
+            if self._log_is_relevant_for_feature(log, node_configs[node_role]):
+                applicable_node_roles.append(node_role)
+        return applicable_node_roles
+
+    def _filter_logs_on_feature_key(self, logs):
+        """Filter from logs all entires that aren't specific to the feature whose logs are desired."""
+        node_configs = self._read_node_configs()
+        filtered_logs = []
+        for log in logs:
+            applicable_node_roles = self._get_node_roles_for_which_feature_is_relevant(log, node_configs)
+            if applicable_node_roles:
+                log["node_roles"] = applicable_node_roles
+                filtered_logs.append(log)
+        assert_that(filtered_logs).is_not_empty()
+        return filtered_logs
+
+    def _filter_logs_on_feature(self, logs):
+        """Filter logs based on whether or not we're testing for a specific feature."""
+        if self.is_feature_specific:
+            return self._filter_logs_on_feature_key(logs)
+        else:
+            return [log for log in logs if not log.get("feature_conditions")]
+
+    def _populate_relevant_logs_for_node_roles(self, logs):
+        """Populate self._relevant_logs with the entries of logs."""
         # When the scheduler is AWS Batch, only keep log that whose config's node_role value is MasterServer, since
         # Batch doesn't have compute nodes in the traditional sense.
-        desired_node_roles = {"MasterServer"} if self.scheduler == "awsbatch" else {"MasterServer", "ComputeFleet"}
+        desired_node_roles = {MASTER_NODE_ROLE_NAME} if self.scheduler == "awsbatch" else set(NODE_ROLE_NAMES)
```

Comment:
Minor: I think this is the only place `NODE_ROLE_NAMES` is used? Maybe declare `NODE_ROLE_NAMES` as a set instead of a list to avoid the set casting here?